File: TransCoder_tokenized_test_set_functions/cpp/NOBLE_INTEGERS_IN_AN_ARRAY_COUNT_OF_GREATER_ELEMENTS_IS_EQUAL_TO_VALUE_1.cpp
int nobleInteger ( int arr [ ], int n ) {
  sort ( arr, arr + n );
  for ( int i = 0;
  i < n - 1;
  i ++ ) {
    if ( arr [ i ] == arr [ i + 1 ] ) continue;
    if ( arr [ i ] == n - i - 1 ) return arr [ i ];
  }
  if ( arr [ n - 1 ] == 0 ) return arr [ n - 1 ];
  return - 1;
}

File: TransCoder_tokenized_test_set_functions/python/NOBLE_INTEGERS_IN_AN_ARRAY_COUNT_OF_GREATER_ELEMENTS_IS_EQUAL_TO_VALUE_1.py
def nobleInteger ( arr ) :
    arr . sort ( )
    n = len ( arr )
    for i in range ( n - 1 ) :
        if arr [ i ] == arr [ i + 1 ] :
            continue
        if arr [ i ] == n - i - 1 :
            return arr [ i ]
    if arr [ n - 1 ] == 0 :
        return arr [ n - 1 ]
    return - 1
